### PR TITLE
docs: fix c.g.c structure

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Making rearrangements to better structure the Storage docs. Googlers see bug below for staged view.

Fixes b/267341907 🦕

### CODE REVIEW

File: docs/acl_guide.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
+ACL
```
Comment (cojenco) [inline docs/acl_guide.rst:1]: Thanks for fixing the structure and moving the guides up. With this update, I plan to modify this guide to include both ACL and IAM. I'll create a separate PR once this is merged in :)

File: docs/index.rst [hunk 2]
```diff
@@ -8,12 +8,31 @@
...
 .. toctree::
   :maxdepth: 2
 
-  storage/modules
+  storage/acl
+  storage/batch
+  storage/blob
+  storage/bucket
+  storage/client
+  storage/constants
```
Comment (cojenco) [inline docs/index.rst:30]: Constants are not rendering on c.g.c. Let's discuss further in our call

**Review by cojenco — APPROVED** [review]
LGTM, thanks Dan!

**Comment by dandhlee** [comment]
Sample issue has been looked into and will be dealt in a separate PR.

File: docs/index.rst (hunk 2 above)
Comment (dandhlee) [reply docs/index.rst:30]: Closing the loop - this is a known issue and I'll look into it separately.

**Pull request merged**